test/suite/test_cursor_bound17.py

```diff
@@ -0,0 +1,89 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wtbound import bound_base
+
+# test_cursor_bound17.py
+# Test the restoring of original primary bounds where setting bounds on some colgroup fails.
+class test_cursor_bound17(bound_base):
+    file_name = 'test_cursor_bound17'
+    use_colgroup = True
+    uri = 'table:'
+
+    key_formats = [
+        ('string', dict(key_format='S')),
+        ('var', dict(key_format='r')),
+        ('int', dict(key_format='i')),
+        ('bytes', dict(key_format='u')),
+        ('composite_string', dict(key_format='SSS')),
+        ('composite_int_string', dict(key_format='iS')),
+        ('composite_complex', dict(key_format='iSru')),
+    ]
+
+    value_formats = [
+        ('string', dict(value_format='S')),
+        ('complex-string', dict(value_format='SS')),
+    ]
+
+    config = [
+        ('inclusive-evict', dict(lower_inclusive=True,upper_inclusive=True,evict=True)),
+        ('no-inclusive-evict', dict(lower_inclusive=False,upper_inclusive=False,evict=True))
+    ]
+
+    direction = [
+        ('prev', dict(next=False)),
+        ('next', dict(next=True)),
+    ]
+
+    scenarios = make_scenarios(key_formats, value_formats, config, direction)
+
+    def test_bound_api(self):
+        cursor = self.create_session_and_cursor()                                                                                                                                                                                                                                        
+
+        # Test default bound api with column groups.
+        self.assertEqual(self.set_bounds(cursor, 40, "lower"), 0) 
+        self.assertEqual(self.set_bounds(cursor, 90, "upper"), 0)
+
+        # Test that original bounds persist even if setting one bound fails. 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 30, "upper"), '/Invalid argument/')
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 95, "lower"), '/Invalid argument/')
+
+        self.assertEqual(self.set_bounds(cursor, 50, "lower"), 0) 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 40, "upper"), '/Invalid argument/')
+        self.cursor_traversal_bound(cursor, 50, 90, None)
+
+        self.assertEqual(self.set_bounds(cursor, 70, "upper"), 0) 
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: self.set_bounds(cursor, 80, "lower"), '/Invalid argument/')
+        self.cursor_traversal_bound(cursor, 50, 70, None)
+
+        #Test successful setting of both bounds.
+        self.assertEqual(self.set_bounds(cursor, 50, "lower"), 0) 
+        self.assertEqual(self.set_bounds(cursor, 70, "upper"), 0) 
+        self.cursor_traversal_bound(cursor, 50, 70, None)
```

Comment:
np: new line at the end of file

Reply:
[0d7073e](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/0d7073eddddb420b84819296e63ad8f41cdf4533)